Testing CI in JAX example

saileshd1402 opened the pull request:
<!--  Thanks for sending a pull request! Here are some tips for you:
1. If this is your first time, check our contributor guidelines: https://www.kubeflow.org/docs/about/contributing
2. To know more about Training Operator, check the developer guide:
    https://github.com/kubeflow/training-operator/blob/master/CONTRIBUTING.md
3. If you want *faster* PR reviews, check how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
-->

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** _(optional, in `Fixes #<issue number>, #<issue number>, ...` format, will close the issue(s) when PR gets merged)_:
Fixes #

**Checklist:**

- [ ] [Docs](https://www.kubeflow.org/docs/components/training/) included if any changes are user facing

andreyvelich commented:
Hi @saileshd1402 @sandipanpanda @mahdikhashan, any updates with the JAX example ?
Since we want to release 1.9 soon, do we want to include this PR as part of the release ?

andreyvelich commented:
cc @Electronic-Waste @kubeflow/wg-training-leads

Electronic-Waste commented:
@andreyvelich  It would be great if we could catch up with the final feature freeze day on Jan 20th. However, we could implement the CI for JAX example with notebooks like https://github.com/kubeflow/training-operator/issues/2246 if the integration test cannot be ready before Jan 20th.

saileshd1402 commented:
I'll try to make the job use lesser resources by the end of today. But if it's not working, then I agree with @Electronic-Waste

mahdikhashan commented:
> Hi @saileshd1402 @sandipanpanda @mahdikhashan, any updates with the JAX example ? Since we want to release 1.9 soon, do we want to include this PR as part of the release ?

i couldn't figure out what could be changed, not yet, regarding the first epoch. i'll spend more time tomorrow.

andreyvelich commented on examples/jax/jax-dist-spmd-mnist/spmd_mnist_classifier_fromscratch.py:
```diff
@@ -0,0 +1,170 @@
+
+    for epoch in range(num_epochs):
+        start_time = time.time()
+        num_batches = 5
```
We can keep this number of batches small.
Was it the main issue why the first epoch takes ~ 1 hour ?

saileshd1402 replied:
Yes, from my understanding this was the reason. Resources was not enough to handle more batches/takes way too long

saileshd1402 commented:
@andreyvelich I have addressed your comment. Seems like CI looks good now, please review when you're available.

cc: @sandipanpanda @mahdikhashan @Electronic-Waste

andreyvelich reviewed (commented):
Thank you for this effort @saileshd1402!
/lgtm
/assign @Electronic-Waste @kubeflow/wg-training-leads @sandipanpanda

Electronic-Waste reviewed (approved):
LGTM. Thanks for your great contribution! @saileshd1402 

/lgtm

andreyvelich reviewed (commented):
Thanks @saileshd1402!
/approve

andreyvelich commented:
/cherry-pick release-1.9
